### ISSUE TITLE
[Theorycrafting] Party Slots

### ISSUE THREAD

**Issue by apple096** [pokeclicker/pokeclicker#814]
Base idea : Party slots. Up to 6.
Sell ability capsules for BP
In an interface similar to evolution items, you can use the capsule on specific Pokemon with outside of battle abilities to unlock the ability. Abilities will only be active while in your party.
Examples:
Harvest: increases berry drop chance.
Flash Fire: increases encounter chance of fire types (if they exist on current route).
Flame Body/Magma Armor: reduce egg hatch time.
Compound Eyes/Pick Up: allows small chance for treasure to drop on routes.
Provide a boost on attack/click attack based on party member's base attack
Basically a "Good ability or good stats"

Will slowly add in more stuffs

**Comment by HLXII** [comment]
Interesting in theory, a bit hard in application.

1. There are 260 canon abilities. It would be difficult to assign a unique usage for each one of those.
    1. You could create/morph the Pokemon abilities to fit Pokeclicker, but that means reassigning abilities to all implemented pokemon.
2. If we don't assign a unique usage for every ability, some Pokemon will have useless abilities, which means spending an ability capsule on them is a waste.
   1. Though this could be mitigated by making the ability capsule not work on these Pokemon, though that means those Pokemon won't ever be used in a party slot.
3. Not sure if it makes sense to lock the Pokemon's ability behind a capsule, since that should already be available to them. Maybe put the price barrier on the party slots?

**Comment by Ultima1990** [comment]
Every Pokémon doesn't need to have an ability, as they also provide multipliers to attack just by being in the party. If you have a big multiplier, no/a useless ability is fine, it's still worth having the pokémon in the party.

**Comment by KrushGames** [comment]
how about maybe 10 abilities that boost things like click attack, egg hatching, oak items, or other things that we give to certain pokemon, and you choose which 6 you want to use

**Comment by Mephistic** [comment]
Love the general idea here, so I made a quick little mock-up:

![party-alpha-small](https://user-images.githubusercontent.com/2694761/95415579-a2bddf80-08fe-11eb-9d24-778806c009d0.gif)

IMO the goal here should be to:
* Give players some strategic options for combat 
* Reward active play

There are likely three things you could gain from putting a pokemon in the active party:
* Stat Boost
  * Maybe a first pass here could just give a modest +10% damage to all mons in the active party - that would give a reason to use it and build out the UI without being overpowered (and without needing to figure out how to map all the in-game abilities to Pokeclicker-suitable abilities in the first pass)
* Abilities - Passive effects you gain just from having something in your party
  * Some abilities have obvious translations (e.g. Magma Armor could speed up hatching)
  * Other abilities could apply with a bit of extra work (e.g. Overgrow/Blaze/Torrent could do bonus damage when there's less than 33% of the timer left since we don't have HP)
* Moves - Active effects that need to be manually triggered (with appropriate cooldowns)
  * Attacking moves could do direct damage (e.g. Flamethrower does 3x DPS in fire damage in one hit)
  * Boosting moves could supplement damage (e.g. Swords Dance could double click damage)
  * Field moves could map to game concepts (e.g. Dig could chip away at squares in the underground, Charge could restore Underground energy, Flash could reveal nearby hidden squares in a dungeon, etc.)

I agree with HLXII that we shouldn't need capsules to unlock abilities - but maybe later on capsules could let you switch between different abilities tied to a single mon (e.g. so Ponyta could use a capsule to switch from Flash Fire to Flame Body)?

It's probably worth considering corresponding nerfs to balance out what we gain from party pokemon (though we probably wouldn't need any for a first pass with a modest stat boost). A few ideas to get started:
* Only active party pokemon can gain XP (or non-party pokemon gain XP at a greatly reduced rate). Even in Kanto before you start collecting achievement boosts, XP isn't an active consideration for dex completion - it only comes into play for grinding out hatches. 
* Boosts are balanced by a corresponding nerf to pokemon outside the party (e.g. Maybe all non-party pokemon contribute Y% less DPS - so using a party lets you gain useful abilities at the cost of passive DPS).

**Comment by Farboo171** [comment]
Closing stale issue since a lot has been added since last discussion, and the main idea drivers are inactive. If still interested, please open a new suggestion with updated ideas.
